### PR TITLE
Feature/healthcheck

### PR DESCRIPTION
**Pull request by DavidSubiros**
## What
- Created this repository
- Copied code from go-ns/s3
- Converted into go module
- Created healthcheck functionality, returning a Check structure. The checker function accepts a bucket name, and tries to do get it - it returns OK Check only if the bucket exists and the S3 client was configured with the same region as the bucket. Otherwise, it returns a CRITICAL structure with the relevant message.
- Created unit tests with mocks created from manual tests with a real S3 client.

## How to review
- Unit tests should pass
- Validate that the S3 client can easily be used by other services relying on this.
- Try to run the unit tests using a real S3 client. (i.e. replace the mock creation in the unit tests with `	s3Cli, _ := s3client.New(<region>)`)

## Who can review
Anyone

### CODE REVIEW

File: healthcheck_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	return fmt.Sprintf("Get /%s/: unsupported protocol scheme \"\"", bucketName)
+}
+
+// bucketExists - mock function for requests for existing buckets
+func bucketExists(bucketName, path string) (io.ReadCloser, error) {
+	return ioutil.NopCloser(strings.NewReader("MockReadCloser")), nil
+}
+
+// bukcetDoesNotExist - mock function for requests with inexistent regions
+func bukcetDoesNotExist(bucketName, path string) (io.ReadCloser, error) {
```
Comment (nshumoogum) [inline healthcheck_test.go]: Misspelt function name

**Comment by nshumoogum** [comment]
`README.md` lacking documentation, see [dp-mongodb README.md](https://github.com/ONSdigital/dp-mongodb/tree/feature/create-health-checker#dp-mongodb)

**Comment by nshumoogum** [comment]
Missing the following files:

* .gitignore
* .github/PULL_REQUEST_TEMPLATE.md
* .travis.yml - see below:
```yml
language: go
script: go test ./...
go:
 - 1.12
 - tip
```

**Comment by nshumoogum** [comment]
Repo setup is wrong for library - This branch should be merged into master (default branch) and develop should be deleted.

I will setup the rules for this repository once you have done this.

File: DESIGN-NOTES.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Design Notes
```
[the remark this answers is not in the document]
Comment (DavidSubiros) [reply DESIGN-NOTES.md]: I saw this file in other repositories. I can remove it if it is not needed.

**Pull request merged**